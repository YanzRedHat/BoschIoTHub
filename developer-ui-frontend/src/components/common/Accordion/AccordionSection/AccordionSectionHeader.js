/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
// SVG Imports
import CaretDown from "images/caretDownIcon.svg";

const ClickProxy = styled.span`
  ${props => !props.sticky && "cursor: pointer"};
  display: inline-flex;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 1;
  align-items: center;
  white-space: nowrap;
  overflow: hidden;
`;

const HeaderContainer = styled.div`
  position: relative;
  z-index: 0;
  display: flex;
  justify-content: space-between;
  height: 4rem;
  align-content: center;
  background: #fff;
  border: 1px solid #ccc;
  font-weight: 500;
  padding: 0 1rem;
  color: #757575;
  transition: box-shadow 0.4s ease-out;
  ${props =>
    props.expanded &&
    `box-shadow: inset 0px -5px 5px 0px rgba(28, 33, 35, 0.14);`};
  ${props =>
    props.disabled &&
    `
    pointer-events: none;
    background: #e4e4e4 !important;
    color: rgba(33, 33, 33, 0.47) !important;
  `}
`;

const ActionButtonsWrapper = styled.span`
  display: inline-flex;
  z-index: 2;
  overflow: visible;
  align-items: center;
  white-space: nowrap;

  svg {
    cursor: pointer;
    width: 1.5rem;
    height: 1.5rem;

    path {
      fill: #757575;
    }
  }
`;

const TitleContainer = styled.span`
  display: inline-flex;
  align-items: center;
  white-space: nowrap;
  overflow: hidden;
`;

const TitleIcon = styled(({ icon, ...props }) =>
  React.cloneElement(icon, props)
)`
  display: inline-block;
  height: 2rem;
  width: 2rem;
  margin-right: 1rem;
  overflow: visible;
  path {
    fill: ${props => {
      if (props.disabled) {
        return "rgba(33, 33, 33, 0.47)";
      }
      return props.theme.accentColor;
    }};
  }
`;

const AccordionSectionHeader = ({
  title,
  icon,
  expanded,
  toggleExpanded,
  children,
  sticky,
  disabled,
  ...other
}) => (
  <div>
    <HeaderContainer expanded={expanded} disabled={disabled} {...other}>
      <ClickProxy onClick={toggleExpanded} sticky={sticky} />
      <TitleContainer>
        {icon ? <TitleIcon icon={icon} disabled={disabled} /> : null}
        {title}
      </TitleContainer>
      <ActionButtonsWrapper>
        {(expanded || sticky) && children}
        {!sticky && !disabled && (
          <CaretDown
            className={
              expanded ? "header-icon-right caret-rotated" : "header-icon-right"
            }
            onClick={toggleExpanded}
          />
        )}
      </ActionButtonsWrapper>
    </HeaderContainer>
  </div>
);

AccordionSectionHeader.propTypes = {
  expanded: PropTypes.bool,
  disabled: PropTypes.bool,
  toggleExpanded: PropTypes.func,
  title: PropTypes.string.isRequired,
  sticky: PropTypes.bool,
  icon: PropTypes.element,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ])
};

export default AccordionSectionHeader;
