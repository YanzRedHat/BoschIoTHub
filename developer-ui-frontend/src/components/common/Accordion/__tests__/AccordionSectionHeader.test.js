/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow, mount } from "enzyme";
import AccordionSectionHeader from "components/common/Accordion/AccordionSection/AccordionSectionHeader.js";
import CaretDown from "images/caretDownIcon.svg";
import styled from "styled-components";

// describe("AccordionSectionHeader", () => {

// });

describe("AccordionSectionHeader", () => {
  let props;
  let mountedAccordionSectionHeader;
  const accordionSectionHeader = () => {
    if (!mountedAccordionSectionHeader) {
      mountedAccordionSectionHeader = mount(
        <AccordionSectionHeader {...props} />
      );
    }
    return mountedAccordionSectionHeader;
  };

  beforeEach(() => {
    props = {
      title: undefined
    };
    mountedAccordionSectionHeader = undefined;
  });

  it("should be defined", () => {
    expect(AccordionSectionHeader).toBeDefined();
  });

  it("should render correctly", () => {
    const tree = shallow(
      <AccordionSectionHeader name="AccordionSectionHeader test" />
    );
    expect(tree).toMatchSnapshot();
  });

  it("always render a div", () => {
    const divs = accordionSectionHeader().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  it.skip("should have a prop children", () => {
    const tree = shallow(
      <AccordionSectionHeader
        name="AccordionSectionHeader test"
        title="Title"
      />
    );
    expect(tree.props().children).toEqual(
      <styled.div name="AccordionSectionHeader test">
        <styled.div />
        <styled.span>Title</styled.span>
        <styled.div>
          <test-file-stub className="header-icon-right" />
        </styled.div>
      </styled.div>
    );
  });
});
