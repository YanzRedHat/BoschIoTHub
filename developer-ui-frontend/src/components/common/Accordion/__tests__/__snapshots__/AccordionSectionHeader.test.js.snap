// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionSectionHeader should render correctly 1`] = `
<div>
  <styled.div
    name="AccordionSectionHeader test"
  >
    <styled.span />
    <styled.span />
    <styled.span>
      <test-file-stub
        className="header-icon-right"
      />
    </styled.span>
  </styled.div>
</div>
`;
